test/jdk/java/awt/Mouse/EnterExitEvents/DragWindowTest.java

```diff
@@ -50,76 +47,64 @@
 import javax.swing.JPanel;
 import javax.swing.SwingUtilities;
 
-import java.util.concurrent.Callable;
-
-import test.java.awt.regtesthelpers.Util;
-
 public class DragWindowTest {
-
     private static volatile int dragWindowMouseEnteredCount = 0;
-    private static volatile int dragWindowMouseReleasedCount = 0;
     private static volatile int buttonMouseEnteredCount = 0;
     private static volatile int labelMouseReleasedCount = 0;
+    private static volatile Point pointToClick;
+    private static volatile Point pointToDrag;
     private static MyDragWindow dragWindow;
     private static JLabel label;
     private static JButton button;
+    private static JFrame frame;
 
     public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
 
-        Robot robot = new Robot();
-        robot.setAutoDelay(100);
+            SwingUtilities.invokeAndWait((Runnable) () -> createAndShowGUI());
 
-        SwingUtilities.invokeAndWait(new Runnable() {
+            robot.delay(250);
+            robot.waitForIdle();
 
-            @Override
-            public void run() {
-                createAndShowGUI();
-            }
-        });
+            SwingUtilities.invokeAndWait(() -> pointToClick = getCenterPoint(label));
 
-        robot.delay(250);
-        robot.waitForIdle();
+            robot.mouseMove(pointToClick.x, pointToClick.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
 
-        Point pointToClick = Util.invokeOnEDT(new Callable<Point>() {
-
-            @Override
-            public Point call() throws Exception {
-                return getCenterPoint(label);
+            if (dragWindowMouseEnteredCount != 1) {
+                throw new RuntimeException("No MouseEntered event on Drag Window!");
             }
-        });
-
-
-        robot.mouseMove(pointToClick.x, pointToClick.y);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.waitForIdle();
 
-        if (dragWindowMouseEnteredCount != 1) {
-            throw new RuntimeException("No MouseEntered event on Drag Window!");
-        }
-
-        Point pointToDrag = Util.invokeOnEDT(new Callable<Point>() {
-
-            @Override
-            public Point call() throws Exception {
+            SwingUtilities.invokeAndWait(() -> {
                 button.addMouseListener(new ButtonMouseListener());
-                return getCenterPoint(button);
-            }
-        });
+                pointToDrag = getCenterPoint(button);
```

Comment:
Can we move this block to l72 right after we get pointToClick...since button is already initialised by now..
It might bring some instability while we are in the midst of drag to do this block and then probably we can remove this additional delay when we have autoDelay

Reply:
I'm in favour of adding listeners when UI is created unless the test logic depends on adding the listener only later.

Reply:
I have moved the listener addition to the createUI method and combined the `getCenterPoint` calls to one `invokeAndWait` block at the start. However, this did not alleviate the need for the additional delays. Removing the delays caused failures on linux and macos. However, when the delays are re-added, the test passes again on all OS's through 3 sets of 100 repeats.

Reply:
What if we move the listener and additionally we reset `buttonMouseEnteredCount ` to 0 before we start the drag `robot.mouseMove(pointToClick.x, pointToClick.y);`
I guess that should not interfere with the test as it is checking for extra mouseEnter event after completion of drag

Reply:
Sure. This may be slightly odd to see a count reset mid-test instead of retaining the original position of adding the listener, but I think this is fine. I re-tested on CI to make sure this still passes, and it looks good on all OS's. Thanks for the suggestion.